### PR TITLE
add patch for srvke-456 backport

### PR DESCRIPTION
**Pull request by lberk**

### CODE REVIEW

**Comment by matzew** [comment]
should we apply the patch ?

File: openshift/patches/015-srvke-456.patch [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+--- a/pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go
++++ b/pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go
+@@ -21,6 +21,7 @@ import (
+ 	"encoding/json"
+ 	"fmt"
+ 
++	"go.uber.org/zap"
+ 	corev1 "k8s.io/api/core/v1"
+ 	"k8s.io/apimachinery/pkg/runtime/schema"
+ 	"knative.dev/eventing/pkg/logging"
```
Comment (matzew) [inline openshift/patches/015-srvke-456.patch:70]: seems to not get added to the file. on my box, @lberk 

Let's update this PR, with the actual changes to the code

/hold

**Comment by lberk** [comment]
/unhold
ok, changed as requested, PTAL @matzew

**Comment by matzew** [comment]
Running `make install` on the branch, gives me:

```
# knative.dev/eventing/pkg/apis/sources/v1beta1
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:102:27: logging.FromContext(ctx).Errorf undefined (type *zap.Logger has no field or method Errorf)
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:104:43: too many arguments in call to resolver.URIFromDestinationV1
	have (context.Context, "knative.dev/pkg/apis/duck/v1".Destination, *SinkBinding)
	want ("knative.dev/pkg/apis/duck/v1".Destination, interface {})
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:106:27: logging.FromContext(ctx).Errorw undefined (type *zap.Logger has no field or method Errorw)
# knative.dev/eventing/pkg/apis/sources/v1beta1
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:102:27: logging.FromContext(ctx).Errorf undefined (type *zap.Logger has no field or method Errorf)
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:104:43: too many arguments in call to resolver.URIFromDestinationV1
	have (context.Context, "knative.dev/pkg/apis/duck/v1".Destination, *SinkBinding)
	want ("knative.dev/pkg/apis/duck/v1".Destination, interface {})
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:106:27: logging.FromContext(ctx).Errorw undefined (type *zap.Logger has no field or method Errorw)
# knative.dev/eventing/pkg/apis/sources/v1beta1
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:102:27: logging.FromContext(ctx).Errorf undefined (type *zap.Logger has no field or method Errorf)
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:104:43: too many arguments in call to resolver.URIFromDestinationV1
	have (context.Context, "knative.dev/pkg/apis/duck/v1".Destination, *SinkBinding)
	want ("knative.dev/pkg/apis/duck/v1".Destination, interface {})
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:106:27: logging.FromContext(ctx).Errorw undefined (type *zap.Logger has no field or method Errorw)
# knative.dev/eventing/pkg/apis/sources/v1beta1
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:102:27: logging.FromContext(ctx).Errorf undefined (type *zap.Logger has no field or method Errorf)
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:104:43: too many arguments in call to resolver.URIFromDestinationV1
	have (context.Context, "knative.dev/pkg/apis/duck/v1".Destination, *SinkBinding)
	want ("knative.dev/pkg/apis/duck/v1".Destination, interface {})
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:106:27: logging.FromContext(ctx).Errorw undefined (type *zap.Logger has no field or method Errorw)
# knative.dev/eventing/pkg/apis/sources/v1beta1
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:102:27: logging.FromContext(ctx).Errorf undefined (type *zap.Logger has no field or method Errorf)
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:104:43: too many arguments in call to resolver.URIFromDestinationV1
	have (context.Context, "knative.dev/pkg/apis/duck/v1".Destination, *SinkBinding)
	want ("knative.dev/pkg/apis/duck/v1".Destination, interface {})
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:106:27: logging.FromContext(ctx).Errorw undefined (type *zap.Logger has no field or method Errorw)
# knative.dev/eventing/pkg/apis/sources/v1beta1
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:102:27: logging.FromContext(ctx).Errorf undefined (type *zap.Logger has no field or method Errorf)
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:104:43: too many arguments in call to resolver.URIFromDestinationV1
	have (context.Context, "knative.dev/pkg/apis/duck/v1".Destination, *SinkBinding)
	want ("knative.dev/pkg/apis/duck/v1".Destination, interface {})
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:106:27: logging.FromContext(ctx).Errorw undefined (type *zap.Logger has no field or method Errorw)
# knative.dev/eventing/pkg/apis/sources/v1beta1
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:102:27: logging.FromContext(ctx).Errorf undefined (type *zap.Logger has no field or method Errorf)
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:104:43: too many arguments in call to resolver.URIFromDestinationV1
	have (context.Context, "knative.dev/pkg/apis/duck/v1".Destination, *SinkBinding)
	want ("knative.dev/pkg/apis/duck/v1".Destination, interface {})
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:106:27: logging.FromContext(ctx).Errorw undefined (type *zap.Logger has no field or method Errorw)
# knative.dev/eventing/pkg/apis/sources/v1beta1
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:102:27: logging.FromContext(ctx).Errorf undefined (type *zap.Logger has no field or method Errorf)
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:104:43: too many arguments in call to resolver.URIFromDestinationV1
	have (context.Context, "knative.dev/pkg/apis/duck/v1".Destination, *SinkBinding)
	want ("knative.dev/pkg/apis/duck/v1".Destination, interface {})
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:106:27: logging.FromContext(ctx).Errorw undefined (type *zap.Logger has no field or method Errorw)
# knative.dev/eventing/pkg/apis/sources/v1beta1
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:102:27: logging.FromContext(ctx).Errorf undefined (type *zap.Logger has no field or method Errorf)
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:104:43: too many arguments in call to resolver.URIFromDestinationV1
	have (context.Context, "knative.dev/pkg/apis/duck/v1".Destination, *SinkBinding)
	want ("knative.dev/pkg/apis/duck/v1".Destination, interface {})
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:106:27: logging.FromContext(ctx).Errorw undefined (type *zap.Logger has no field or method Errorw)
# knative.dev/eventing/pkg/apis/sources/v1beta1
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:102:27: logging.FromContext(ctx).Errorf undefined (type *zap.Logger has no field or method Errorf)
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:104:43: too many arguments in call to resolver.URIFromDestinationV1
	have (context.Context, "knative.dev/pkg/apis/duck/v1".Destination, *SinkBinding)
	want ("knative.dev/pkg/apis/duck/v1".Destination, interface {})
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:106:27: logging.FromContext(ctx).Errorw undefined (type *zap.Logger has no field or method Errorw)
# knative.dev/eventing/pkg/apis/sources/v1beta1
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:102:27: logging.FromContext(ctx).Errorf undefined (type *zap.Logger has no field or method Errorf)
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:104:43: too many arguments in call to resolver.URIFromDestinationV1
	have (context.Context, "knative.dev/pkg/apis/duck/v1".Destination, *SinkBinding)
	want ("knative.dev/pkg/apis/duck/v1".Destination, interface {})
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:106:27: logging.FromContext(ctx).Errorw undefined (type *zap.Logger has no field or method Errorw)
go build -o /home/matzew/go/bin/mtbroker_ingress ./cmd/mtbroker/ingress/
# knative.dev/eventing/pkg/apis/sources/v1beta1
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:102:27: logging.FromContext(ctx).Errorf undefined (type *zap.Logger has no field or method Errorf)
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:104:43: too many arguments in call to resolver.URIFromDestinationV1
	have (context.Context, "knative.dev/pkg/apis/duck/v1".Destination, *SinkBinding)
	want ("knative.dev/pkg/apis/duck/v1".Destination, interface {})
pkg/apis/sources/v1beta1/sinkbinding_lifecycle.go:106:27: logging.FromContext(ctx).Errorw undefined (type *zap.Logger has no field or method Errorw)
make: *** [Makefile:17: install] Error 2
```

**Comment by lberk** [comment]
> Running `make install` on the branch, gives me:

Yep, thanks, saw the prow output, fixing now

**Comment by lberk** [comment]
@matzew I don't think this is going to work for 0.17 -- that era `kantive.dev/pkg` doesn't have a `subResourceReconciler` set of functions which is needed to mirror the resolver status out.

**Comment by matzew** [comment]
OK thanks for checking, @lberk  

I think we could live w/ it when it would land on the 0.18.x APIs - Is that possible?

**Comment by lberk** [comment]
@matzew I've got the release-0.18 backport open upstream -- when that lands do you want me to update our downstream branch? https://github.com/knative/eventing/pull/4401

**Comment by matzew** [comment]
> @matzew I've got the release-0.18 backport open upstream -- when that lands do you want me to update our downstream branch? knative#4401

Ah - cool. Once we land that upstream, I don't think we need to backport to here.

I'd like to create a newer/fresher `release-v0.18.z` branch anyways, so no need to waste time

**Pull request closed without merging**